test/reinit_test.jl

```diff
@@ -44,3 +44,29 @@ solve!(integrator)
 
 @test u == integrator.sol.u
 @test t == integrator.sol.t
+
+#callback test
+f(t,u) = 2.0*t-2.0
+u0=0.0
+tspan = (0.0,4.0)
+prob = ODEProblem(f,u0,tspan)
+function condition(t,u,integrator) # Event when event_f(t,u) == 0
+  u
+end
+function affect!(integrator)
+  integrator.u = 0.7
+end
+cb = ContinuousCallback(condition,affect!)
+integrator = init(prob,Tsit5();dt=1//2^(4),callback = cb)
+dt = integrator.dt
+solve!(integrator)
+
+u = copy(integrator.sol.u)
+t = copy(integrator.sol.t)
+
+reinit!(integrator)
```

Comment:
But saving callback can only save the values when callback is called. It cannot change the initial condition. So we need the Continuous Callback for initializing.

Reply:
`SavingCallback`'s reinit has to reset the `saveat` values, so it will only save at the `saveat` values the second time if `reinit` on a callback works as expected.